cmd/kyma/provision/gardener/cmd.go

```diff
@@ -0,0 +1,143 @@
+package gardener
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/kyma-project/cli/internal/kube"
+
+	hf "github.com/kyma-incubator/hydroform"
+	"github.com/kyma-incubator/hydroform/types"
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/cli/internal/files"
+	"github.com/spf13/cobra"
+)
+
+type command struct {
+	opts *Options
+	cli.Command
+}
+
+//NewCmd creates a new minikube command
+func NewCmd(o *Options) *cobra.Command {
+	c := command{
+		Command: cli.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "gardener",
+		Short: "Provisions a Gardener cluster.",
+		Long:  `Use this command to provision Gardener for Kyma installation.`,
+		RunE:  func(_ *cobra.Command, _ []string) error { return c.Run() },
+	}
+
+	cmd.Flags().StringVarP(&o.Name, "name", "n", "", "Name of the cluster to provision.")
+	cmd.Flags().StringVarP(&o.Project, "project", "p", "", "Name of the Gardener Project where to provision the cluster in.")
+	cmd.Flags().StringVarP(&o.CredentialsFile, "credentials", "c", "", "Path to the Gardener service account kubueconfig file.")
+	cmd.Flags().StringVar(&o.TargetProvider, "target-provider", "gcp", "Specify the cloud provider that Gardener should use to create the cluster.")
+	cmd.Flags().StringVarP(&o.Secret, "secret", "s", "", "Name of the Gardener secret to access the target provider.")
+	cmd.Flags().StringVarP(&o.KubernetesVersion, "kube-version", "k", "1.15.4", "Kubernetes version of the cluster to provision.")
+	cmd.Flags().StringVarP(&o.Location, "location", "l", "europe-west3", "Location of the cluster to provision.")
```

Comment:
We can rename `location` to `region` to be consistent with the cloud providers.

Reply:
done thanks!